src/process/data-provider.js

```diff
@@ -12,16 +12,42 @@ const enrichedDataQuery = `
     and timestamp >= now() - interval  '${to}'
 `;
 
+const archiveEnrichedData =
+    `select * from enriched_data 
+    where timestamp <= now()- interval '${to}' 
+    order by timestamp
+limit $1
+ offset $2`;
+
 const parseTimestamp = (timestamp) => DateTime.fromJSDate(timestamp);
 const mapRow = (row) => {
     return (row) ? new Measurement(row.station, row.pm10, row.pm25, parseTimestamp(row.timestamp)) : null;
 }
 
-const getMeasurement = async (stationName) => {
+const mapRows = (rows) => {
+    let measurements = [];
+
+    for (const row of rows) {
+        (row) ? measurements.push(new Measurement(row.station, row.pm10, row.pm25, parseTimestamp(row.timestamp))) : null;
```

Comment:
całość do wyjebania razem z pętlą.
Po co masz metodą mapRow?
Albo robisz:
```
for (const row of rows) {
    const measurement = mapRow(row);
    if (measurement) measurements.push(measurement);
}
```
albo od razu robisz to używając operacji takich jak mapa:
```
const measurements = rows.map(mapRow).filter(measurement => measurement !== null)
```